lib/phoenix_live_view/html_algebra.ex

```diff
@@ -437,7 +437,7 @@ defmodule Phoenix.LiveView.HTMLAlgebra do
 
           # delimiter for normal strings are "\""
           _ ->
-            ~s(#{attr}="#{string}")
+            ~s(#{attr}="#{Phoenix.HTML.html_escape(string) |> Phoenix.HTML.safe_to_string()}")
```

Comment:
The problem is that we cannot simply escape `"`, because that wouldn't result in a valid attribute:

```elixir
~H"""
<div data-foo="{\"tag\": \"<something>\"}"></div>
"""
```

And I don't see the benefit of manually defining an escape function which then basically does what Phoenix.HTML's escape does 🤔

Reply:
The other option would be to simply not replace `attr={"string"}` with `attr="string"` when there are double quotes inside the string. That may actuall be preferable.

Reply:
@josevalim I changed the code to not replace expressions that are strings with quotes.